source/io/module_javaScriptInvoke.js

```diff
@@ -452,32 +392,28 @@ var assignJavaScriptInvoke;
 
         Module.javaScriptInvoke.jsJavaScriptInvoke = function (
             occurrencePointer,
-            functionNamePointer,
+            functionNameTypeRef,
+            functionNameDataRef,
             returnPointer,
             parametersPointer,
             parametersCount,
             isInternalFunction,
             errorTypeRef,
             errorDataRef) {
             var errorValueRef = Module.eggShell.createValueRef(errorTypeRef, errorDataRef);
+            var functionNameValueRef = Module.eggShell.createValueRef(functionNameTypeRef, functionNameDataRef);
+            var functionName = Module.eggShell.readString(functionNameValueRef);
+            var parameters = [];
 
-            var newError = {
-                status: false,
-                code: ERRORS.NO_ERROR.CODE,
-                source: ERRORS.NO_ERROR.MESSAGE
-            };
+            var returnValueRef = createValueRefFromPointerArray(returnPointer, 0);
+            if (isInternalFunction) {
+                parameters = parameters.concat(returnValueRef);
+            }
 
-            var errorMessage = '';
-            var functionName = Module.eggShell.dataReadString(functionNamePointer);
-            var parameters = undefined;
             try {
-                parameters = createJavaScriptParametersArray(parametersPointer, parametersCount);
+                parameters = parameters.concat(createJavaScriptParametersArray(isInternalFunction, parametersPointer, parametersCount));
```

Comment:
can do `Array.prototype.push.apply(parameters, createJavaScriptParametersArray(isInternalFunction, parametersPointer, parametersCount))` to avoid extra array copies